### DISCUSSION TITLE
equal with not Equatable?

### DISCUSSION THREAD

**Discussion by the-yuyut** [Quick/Nimble#987] [Ideas]
For me, I have trouble asserting equal to [String: Any], 
So I use a wrapper to use NSDictionary to do that, I hope that the `internal func equal(_:by:) -> Predicate<T>` is not internal, so that I don't have to copy past the code as an extension of my app.

**Comment by younata** [comment]
hey @ytyubox! Thanks for the feedback!

~~Yeah, making `equal(_:by:)` public sounds like a good idea. I'll do that + write up some documentation on using it.~~

My apologies! On second thought, I'm not so certain this is such a good idea. We'd support you if you wanted to write a custom matcher that uses NSDictionary under the hood to do this, but I'm not so certain this is a great idea as a general matcher. Generally, your API shouldn't be returning or handling values of type [String: Any].

I'm open to discussion if you have additional reasons for doing this, but for now, I don't think Nimble should expose `equal(_:by:)`.
